src/components/modals/ImportWorkspaceFileModal.svelte

```diff
@@ -23,21 +27,37 @@
 
   const dispatch = createEventDispatcher<{
     close: void;
-    confirm: { files: FileList; targetDirectory: string };
+    confirm: {
+      convertedFileExtension: string;
+      files: FileList;
+      keepOriginalFiles: boolean;
+      shouldConvert: boolean;
+      targetDirectory: string;
+    };
   }>();
 
-  let targetPath: string = joinPath([currentWorkspace.name, startingPath]);
-  let targetDirectory: string = startingPath;
-  let targetFilename: string = '';
+  let targetDirectory: string = joinPath([currentWorkspace?.name ?? '', startingPath]);
   let saveButtonDisabled: boolean = false;
   let filesToUpload: FileList;
+  let keepOriginalFiles: boolean = false;
+  let shouldConvert: boolean = false;
+  let convertedFileExtension: string = '.seqN.txt';
+  let numberOfConvertableFiles: number = 0;
 
   $: {
-    const { filename, path } = separateFilenameFromPath(targetPath);
-    targetDirectory = path;
-    targetFilename = filename;
+    saveButtonDisabled = filesToUpload?.length === 0;
+    numberOfConvertableFiles = 0;
+    for (let i = 0; i < (filesToUpload?.length ?? 0); i++) {
+      const file = filesToUpload[i];
+      const extension = file.name.replace(/^(?:[^.]+)(\..+)?$/, '$1');
+      if (
+        extension &&
+        outputLanguageExtensions.findIndex(fileExtension => extension === `.${fileExtension.replace(/^\./, '')}`) > -1
+      ) {
```

Comment:
```suggestion
      if (outputLanguageExtensions.findIndex(fExt => file.name.toLowerCase().endsWith(fExt.toLowerCase()))) {
```

MIght be a learning opportunity for me about why this is necessary, but can we save all the regex here and just check `endsWith`? It seems like this might also fail on the case of a file with periods before the extension we're looking for? (e.g., `my.file.seq.json`), ~and incorrectly match extensions that differ in periods (e.g., match `.seqjson` to `.seq.json`).~ Correction: I missed the anchor in the second regex; that's just to handle whether the adaptation does or doesn't provide the leading period in the file extension.

Reply:
You are correct. This was done intentionally. `endsWith` does not take a regex as an argument. Passing the expected extension would result in potentially unintended matches. The intent is that an output file extension of `.json` should not match `.seq.json`.

Reply:
We hashed this one out for a while and I seem to have generally convinced Bryan that `endsWith` has acceptable behavior here, but just to enumerate some of the examples we went through and what behavior we'd see for each:

| Filename          | Extension   | Regex match? | `endsWith` match? | Expect to match?        |
| :---------------- | :---------- | :----------- | :---------------- | :---------------------- |
| `file.seq.json`   | `.json`     | No           | Yes               | We disagree             |
| `file.seq.json`   | `.seq.json` | Yes          | Yes               | Yes                     |
| `file.a.seq.json` | `.seq.json` | No           | Yes               | Yes (Clipper does this) |
| `file.seqjson`    | `.seq.json` | No           | No                | No                      |